src/backend/distributed/executor/multi_server_executor.c

```diff
@@ -56,9 +56,11 @@ JobExecutorType(MultiPlan *multiPlan)
 	if (taskCount > 0)
 	{
 		Task *firstTask = (Task *) linitial(workerTaskList);
+		TaskType taskType = firstTask->taskType;
```

Comment:
This is code path is still applicable. If query is router executable (single shard query) but can't detect it in router planner, this existing path can still catch it.

Doing this in an automated fashion is implemented by @marcocitus earlier. But we haven't integrated it into code path. @ozgune suggested we do it after we are clear of router planner implementation.